extensions/sql-migration/src/models/stateMachine.ts

```diff
@@ -394,12 +394,95 @@ export class MigrationStateModel implements Model, vscode.Disposable {
+						'sessionId': this._sessionId,
+						'recommendedSku': JSON.stringify(resultItem?.targetSku)
+					},
+					{}
+				);
+			});
+
+			// Send Instance requirements used for calculating recommendations
+			sendSqlMigrationActionEvent(
+				TelemetryViews.SkuRecommendationWizard,
+				TelemetryAction.GetInstanceRequirements,
+				{
+					'sessionId': this._sessionId,
+					'performanceDataSource': this._skuRecommendationPerformanceDataSource,
+					'cpuRequirementInCores': this._skuRecommendationResults?.recommendations?.instanceRequirements?.cpuRequirementInCores?.toString(),
+					'dataStorageRequirementInMB': this._skuRecommendationResults?.recommendations?.instanceRequirements?.dataStorageRequirementInMB?.toString(),
+					'logStorageRequirementInMB': this._skuRecommendationResults?.recommendations?.instanceRequirements?.logStorageRequirementInMB?.toString(),
+					'memoryRequirementInMB': this._skuRecommendationResults?.recommendations?.instanceRequirements?.memoryRequirementInMB.toString(),
+					'dataIOPSRequirement': this._skuRecommendationResults?.recommendations?.instanceRequirements?.dataIOPSRequirement?.toString(),
+					'logIOPSRequirement': this._skuRecommendationResults?.recommendations?.instanceRequirements?.logIOPSRequirement?.toString(),
+					'ioLatencyRequirementInMs': this._skuRecommendationResults?.recommendations?.instanceRequirements?.ioLatencyRequirementInMs?.toString(),
+					'ioThroughputRequirementInMBps': this._skuRecommendationResults?.recommendations?.instanceRequirements?.ioThroughputRequirementInMBps?.toString(),
+					'tempDBSizeInMB': this._skuRecommendationResults?.recommendations?.instanceRequirements?.tempDBSizeInMB?.toString(),
+					'aggregationTargetPercentile': this._skuRecommendationResults?.recommendations?.instanceRequirements?.aggregationTargetPercentile?.toString(),
+					'numberOfDataPointsAnalyzed': this._skuRecommendationResults?.recommendations?.instanceRequirements?.numberOfDataPointsAnalyzed?.toString(),
+					'databaseLevelRequirements': JSON.stringify(this._skuRecommendationResults?.recommendations?.instanceRequirements?.databaseLevelRequirements?.map(i => {
+						return {
+							cpuRequirementInCores: i.cpuRequirementInCores,
+							dataIOPSRequirement: i.dataIOPSRequirement,
+							logIOPSRequirement: i.logIOPSRequirement,
+							ioLatencyRequirementInMs: i.ioLatencyRequirementInMs,
+							ioThroughputRequirementInMBps: i.ioThroughputRequirementInMBps,
+							dataStorageRequirementInMB: i.dataStorageRequirementInMB,
+							logStorageRequirementInMB: i.logStorageRequirementInMB,
+							databaseName: hashString(i.databaseName),
+							memoryRequirementInMB: i.memoryRequirementInMB,
+							cpuRequirementInPercentageOfTotalInstance: i.cpuRequirementInPercentageOfTotalInstance,
+							numberOfDataPointsAnalyzed: i.numberOfDataPointsAnalyzed,
+							fileLevelRequirements: i.fileLevelRequirements?.map(file => {
+								return {
+									fileName: hashString(file.fileName),
```

Comment:
You're hashing this, which is good. But do you really need this? Generally we prefer that unless you actually have a use case for a data point that you just leave it out.

Reply:
Thanks Charles for pointing this.. we don't actually need file names, filetype is important. Let me remove in next commit.